pkg/build/builder/cmd/scmauth/passwd.go

```diff
@@ -0,0 +1,75 @@
+package scmauth
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	UsernamePasswordName = "passwd"
+	UserPassGitConfig    = `# username and password git config
+[url "%[1]s"]
+   insteadOf = %[2]s
+`
+)
+
+// SSHPrivateKey implements SCMAuth interface for using SSH private keys.
+type UsernamePassword struct {
+	SourceURL url.URL
+}
+
+// Setup creates a wrapper script for SSH command to be able to use the provided
+// SSH key while accessing private repository.
+func (u UsernamePassword) Setup(baseDir string) error {
```

Comment:
Excellent idea

On Aug 26, 2015, at 11:16 PM, Jordan Liggitt <notifications@github.com>
wrote:

In pkg/build/builder/cmd/scmauth/passwd.go
<https://github.com/openshift/origin/pull/4387#discussion_r38056644>:

> +const (
> +	UsernamePasswordName = "passwd"
> +	UserPassGitConfig    = `# username and password git config
> +[url "%[1]s"]
> +   insteadOf = %[2]s
> +`
> +)
> +
> +// SSHPrivateKey implements SCMAuth interface for using SSH private keys.
> +type UsernamePassword struct {
> +	SourceURL url.URL
> +}
> +
> +// Setup creates a wrapper script for SSH command to be able to use the provided
> +// SSH key while accessing private repository.
> +func (u UsernamePassword) Setup(baseDir string) error {

also allow pulling the username from the URL, e.g. "
https://myuser@github.com/openshift/origin.git", and populating the
password from the secret

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4387/files#r38056644>.